File: scripts/run-client-hunting.ts
#!/usr/bin/env npx tsx
/**
 * Jarvis Client Hunting CLI
 * Usage: npx tsx scripts/run-client-hunting.ts --brand bulletseye [--brief "..."] [--limit 5]
 */

import "./load-env-local";
import { runClientHuntingPipeline } from "@/faos_core/pipelines/client-hunting-pipeline";
import { HUNTING_BRAND_IDS, huntingBrandsConfig } from "@/faos_core/config/brands";

function arg(flag: string): string | undefined {
  const idx = process.argv.indexOf(flag);
  if (idx >= 0 && process.argv[idx + 1]) return process.argv[idx + 1];
  return undefined;
}

async function main() {
  if (process.argv.includes("--list")) {
    console.log(JSON.stringify(huntingBrandsConfig(), null, 2));
    return;
  }

  const brand = arg("--brand") || arg("-b");
  const brief = arg("--brief");
  const limitRaw = arg("--limit");
  const limit = limitRaw ? Number(limitRaw) : undefined;

  if (brand && !HUNTING_BRAND_IDS.includes(brand as (typeof HUNTING_BRAND_IDS)[number])) {
    console.error(
      `Unknown brand "${brand}". Use one of: ${HUNTING_BRAND_IDS.join(" | ")}`
    );
    process.exit(2);
  }

  const result = await runClientHuntingPipeline({ brand, brief, limit });
  const delivery = {
    brand: result.brand,
    task_id: result.task_id,
    prospects_targeted: result.prospects_targeted,
    content_assets: result.content_assets,
    action_items: result.action_items,
    status: result.status,
    brand_id: result.brand_id,
    hub: result.hub,
    lead_agent_id: result.lead_agent_id,
    namespace: result.namespace,
    version: result.version,
    ok: result.ok,
    error: result.error,
  };

  console.log(JSON.stringify(delivery, null, 2));
  if (!result.ok) process.exit(1);
}

main().catch((err) => {
  console.error(err instanceof Error ? err.message : err);
  process.exit(1);
});

File: scripts/run_client_hunting.py
#!/usr/bin/env python3
"""Jarvis Client Hunting CLI (Python entry).

Delegates to the TS pipeline so brand locks + Hermes routing stay single-sourced.

Usage:
  python scripts/run_client_hunting.py --brand fmk_agency
  python scripts/run_client_hunting.py --brand bulletseye --brief "Meta audit outreach"
  python scripts/run_client_hunting.py --brand fmk_wig --limit 4
"""

from __future__ import annotations

import argparse
import shutil
import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
BRANDS = ("fmk_agency", "bulletseye", "fmk_wig")


def main() -> int:
    parser = argparse.ArgumentParser(description="FAOS Jarvis client hunting funnel")
    parser.add_argument("--brand", choices=BRANDS, required=False, help="Target brand")
    parser.add_argument("--brief", default="", help="Optional acquisition brief")
    parser.add_argument("--limit", type=int, default=5, help="Prospect count (1-12)")
    parser.add_argument("--list", action="store_true", help="List brand config")
    args = parser.parse_args()

    npx = shutil.which("npx")
    if not npx:
        print("npx not found — install Node 22+ to run the hunting pipeline", file=sys.stderr)
        return 1

    cmd = [npx, "tsx", str(ROOT / "scripts" / "run-client-hunting.ts")]
    if args.list:
        cmd.append("--list")
    if args.brand:
        cmd.extend(["--brand", args.brand])
    if args.brief:
        cmd.extend(["--brief", args.brief])
    if args.limit:
        cmd.extend(["--limit", str(args.limit)])

    env = dict(**{k: v for k, v in __import__("os").environ.items()})
    # Prefer local .env.local via scripts/load-env-local.ts (loaded by TS entry)
    proc = subprocess.run(cmd, cwd=str(ROOT), env=env)
    return int(proc.returncode)


if __name__ == "__main__":
    raise SystemExit(main())
